app/controllers/settings_controller.rb

```diff
@@ -55,6 +55,11 @@ def using_incoming_sms_token_message
   end
 
   private
+    def authorize!(*args)
+      super
+      require_recent_login
+    end
```

Comment:
Hmm, just noticed this. I am definitely puzzled by this so I think we need a big fat comment here.